client/my-sites/checkout/composite-checkout/lib/is-wpcom-product-renewal.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * External dependencies
+ */
+import type { ResponseCartProduct } from '@automattic/shopping-cart';
+
+export default function isWpComProductRenewal( product: ResponseCartProduct ): boolean {
```

Comment:
I wonder if this would fit neatly into either `packages/calypso-products` (which already has quite a few "is..." functions) or, failing that, `packages/wpcom-checkout`? Both packages support TypeScript. I only say this because we're trying to move all checkout dependencies out of calypso so this will have to go somewhere eventually. Also

Reply:
Thanks for taking a look, @sirbrillig!

I would love to push this up into a package, but I am honestly not sure about two things:
1. Is the `product?.extra?.purchaseType === 'renewal'` a valid check for all products, especially those not for WordPress.com? I kept this scoped to WordPreass.com products because that's the area where I believe this is true!
2. The current logic depends on the `ResponseCartProduct` type, which is defined in `packages/shopping-cart` - the `calypso-products`. I am not sure it make sense to create a similar type for `calypso-products`, or whether we can expect the current code to work as expected everywhere `calypso-products` is used. We can add this logic to `packages/wpcom-checkout` fairly easily, as it already depends on `packages/shopping-cart`.

Any and all thoughts would be welcome!

Reply:
> Is the product?.extra?.purchaseType === 'renewal' a valid check for all products, especially those not for WordPress.com

It's true for all products that pass through the WordPress.com shopping cart system, which also includes Jetpack products. It's still a fair question because "product" can mean many things (eg: products endpoint products, receipt item products, shopping cart item products, etc.) but if you're talking about a wpcom shopping cart product, then this is true for all of them.

> The current logic depends on the ResponseCartProduct type, which is defined in packages/shopping-cart

True, as you said, `packages/wpcom-checkout` [already has a dependency on `packages/shopping-cart`](https://github.com/Automattic/wp-calypso/blob/84f111481d619bd932563dad253795a019a162e6/packages/wpcom-checkout/package.json#L42), so that's a pretty safe place to put it. (I suspect `packages/calypso-products` will eventually end up with that dependency too, but you're right that adding it here might be more than this PR might want to be doing.) Both packages currently are just a way to extract and organize logic out of calypso, with only calypso as a consumer. The idea is that moving things into those packages will help untangle the dependency trees a bit and will also make them available if, for example, we want to try implementing checkout into wp-admin.